basic_ui/settings/Softwareupdate.qml

```diff
@@ -158,6 +158,7 @@ Item {
             mouseArea.onClicked: {
                 if (updateAvailable) {
                     console.debug("Updating the app")
+                    // TODO create a update servcie class instead of launching hard coded shell scripts from QML
```

Comment:
Software update also should be handled on the C++ side. Possible using QNetworkAccessManager.

Reply:
Absolutely. This could be implemented in #346 Hardware device abstraction. I didn't want to mix it with the wifi rewrite.